src/main/java/org/broadinstitute/hellbender/tools/walkers/sv/SVAnnotate.java

```diff
@@ -0,0 +1,479 @@
+            consequence = annotateCopyNumberVariant(variantInterval,transcript);
+        } else if (svType.equals(StructuralVariantAnnotationType.INV)) {
+            consequence = annotateInversion(variantInterval, transcript);
+        } else if (svType.equals(StructuralVariantAnnotationType.CTX)) {
+            consequence = annotateTranslocation(variantInterval, transcript);
+        } else if (svType.equals(StructuralVariantAnnotationType.BND)) {
+            consequence = annotateBreakend(variantInterval, transcript);
+        }
+
+        if (consequence != null) {
+            updateVariantConsequenceDict(variantConsequenceDict, consequence, transcript.getGeneName());
+        }
+    }
+
+    /**
+     Annotate promoter overlaps and return a boolean: true if there are any promoter overlaps, false if not
+     */
+    private boolean annotatePromoterOverlaps(final SimpleInterval variantInterval, final Map<String, Set<String>> variantConsequenceDict) {
+        boolean anyPromoterOverlaps = false;
+        final Set<FullBEDFeature> promotersForVariant = promoterOverlapDetector.getOverlaps(variantInterval);
+        final Set<String> codingAnnotationGenes = new HashSet<>();
+        variantConsequenceDict.values().forEach(codingAnnotationGenes::addAll);
+        for (FullBEDFeature promoter : promotersForVariant) {
+            if (!codingAnnotationGenes.contains(promoter.getName())) {
+                updateVariantConsequenceDict(variantConsequenceDict, GATKSVVCFConstants.PROMOTER, promoter.getName());
+                anyPromoterOverlaps = true;
+            }
+        }
+        return anyPromoterOverlaps;
+    }
+
+    private void annotateNonCodingOverlaps(final SimpleInterval variantInterval, final Map<String, Set<String>> variantConsequenceDict) {
+        final Set<FullBEDFeature> nonCodingFeaturesForVariant = nonCodingOverlapDetector.getOverlaps(variantInterval);
+        for (FullBEDFeature feature : nonCodingFeaturesForVariant) {
+            String consequence = GATKSVVCFConstants.NONCODING_BREAKPOINT;
+            if (variantSpansFeature(variantInterval, new SimpleInterval(feature))) {
+                consequence = GATKSVVCFConstants.NONCODING_SPAN;
+            }
+            updateVariantConsequenceDict(variantConsequenceDict, consequence, feature.getName());
+        }
+    }
+
+
+    protected static void annotateNearestTranscriptionStartSite(final SimpleInterval variantInterval, final Map<String, Set<String>> variantConsequenceDict, SVIntervalTree<String> transcriptionStartSiteTree, int maxContigLength, int variantContigID) {
+        // TODO: keep all nearest TSS for dispersed CPX / CTX or choose closest?
+        // TODO: will start < end ever? Shouldn't at this point in the pipeline
+        SVIntervalTree.Entry<String> nearestBefore = transcriptionStartSiteTree.max(new SVInterval(variantContigID, variantInterval.getStart(), variantInterval.getEnd()));
```

Comment:
It's not super important here--I don't think it will have much of an impact on performance--but just in general best not to repeat operations unnecessarily:  Pull out the creation of the new SVInterval into a local variable so you don't repeat it.

Reply:
Hmm.  Now that I think it over, I believe you want to take the min on (contig, start, start) and the max on (contig, end, end) rather than on (contig, start, end) as you're currently doing.